QRCoder/QRCodeGenerator.ModulePlacer.cs

```diff
@@ -160,29 +162,30 @@ public static int MaskCode(QRCodeData qrCode, int version, List<Rectangle> block
                     // Select the pattern with the lowest score, indicating better QR code readability.
                     if (!selectedPattern.HasValue || patternScore > score)
                     {
-                        selectedPattern = pattern.Key;
+                        selectedPattern = maskPattern;
                         patternScore = score;
                     }
                 }
 
                 // Apply the best mask pattern to the actual QR code.
+                var selectedPatternFunc = MaskPattern.Patterns[selectedPattern.Value];
```

Comment:
Why `selectedPattern.Value`? Shouldn't simply `selectedPattern` work, too, as selectedPattern should be an int (due to `selectedPattern = maskPattern`)?

Reply:
it's an `int?` since it's `null` initially.  it will set a value on the first pass through the loop, so it will never be null here.

Reply:
Changed to `int`, by initializing the initial score to `int.MaxValue` so all patterns score better than the variable's initial values

Reply:
Sorry, my fault. I was to focused on the diff view and did not check the type of selectedPattern. Then assumed it must be int because maskPattern is of type int.